tests/test_everflow_testbed.py

```diff
@@ -0,0 +1,664 @@
+DUT_TMP_DIR = os.path.join('tmp', os.path.basename(BASE_DIR))
+FILES_DIR = os.path.join(BASE_DIR, 'files')
+TEMPLATE_DIR = os.path.join(BASE_DIR, 'templates')
+
+EVERFLOW_TABLE_RULE_CREATE_TEMPLATE = 'acl_rule_persistent.json.j2'
+EVERFLOW_TABLE_RULE_CREATE_FILE = 'acl_rule_persistent.json'
+EVERFLOW_TABLE_RULE_DELETE_FILE = 'acl_rule_persistent-del.json'
+DUT_RUN_DIR = '/home/admin/everflow_tests'
+
+@pytest.fixture(scope="module", autouse=True)
+def copy_acstests_directory(ptfhost):
+    """ Fixture which copies the ptftests directory to the PTF host. This fixture
+        is scoped to the module, as it only needs to be performed once before
+        the first test is run. It does not need to be run before each test.
+        We also set autouse=True to ensure this fixture gets instantiated before
+        the first test runs, even if we don't explicitly pass it to them, and since
+        there is no return value, there is no point in passing it into the functions.
+    """
+    ptfhost.copy(src="acstests", dest="/root")
+
+@pytest.fixture(scope='module')
+def setup_info(duthost, testbed):
+    """
+    setup fixture gathers all test required information from DUT facts and testbed
+    :param duthost: DUT host object
+    :param testbed: Testbed object
+    :return: dictionary with all test required information
+    """
+
+    tor_ports = []
+    spine_ports = []
+    port_channels = []
+
+    # gather ansible facts
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    #gather switch capability facts
+    switch_capability_facts = duthost.switch_capabilities_facts()['ansible_facts']
+    #gather host facts
+    host_facts = duthost.setup()['ansible_facts']
+
+    # get the list of TOR/SPINE ports
+    for dut_port, neigh in mg_facts['minigraph_neighbors'].items():
+        if 'T0' in neigh['name']:
+            tor_ports.append(dut_port)
+        elif 'T2' in neigh['name']:
+            spine_ports.append(dut_port)
+
+    # get the list of port channels
+    port_channels = mg_facts['minigraph_portchannels']
+
+    test_mirror_v4 = True if switch_capability_facts['switch_capabilities']['switch']['MIRROR'] == 'true' else False
+    test_mirror_v6 = True if switch_capability_facts['switch_capabilities']['switch']['MIRRORV6'] == 'true' else False
+    test_ingress_mirror_on_ingress_acl = True if 'MIRROR_INGRESS_ACTION' in switch_capability_facts['switch_capabilities']['switch']['ACL_ACTIONS|INGRESS'] else False
+    test_ingress_mirror_on_egress_acl =  True if 'MIRROR_INGRESS_ACTION' in switch_capability_facts['switch_capabilities']['switch']['ACL_ACTIONS|EGRESS'] else False
+    test_egress_mirror_on_egress_acl =  True if 'MIRROR_EGRESS_ACTION' in switch_capability_facts['switch_capabilities']['switch']['ACL_ACTIONS|EGRESS'] else False
+    test_egress_mirror_on_ingress_acl = True if 'MIRROR_EGRESS_ACTION' in switch_capability_facts['switch_capabilities']['switch']['ACL_ACTIONS|INGRESS'] else False
```

Comment:
These checks are all pretty redundant, you can just use the boolean expression. For example, for `test_mirror_v4`:
```
test_mirror_v4 = switch_capability_facts['switch_capabilities']['switch']['MIRROR'] == 'true'
```

Reply:
Thnaks. 

> can you check on the mlnx platform?

Verfied on Mellanox Platform for Ingress Mirror/Ingress ACL on 201811 image. All Applicable 5 test cases are passing.

py.test  -vvv --inventory veos.vtb  --host-pattern all --user admin  --testbed vms11-2-t1 --testbed_file vtestbed.csv  --disable_loganalyzer --show-capture stdout --junitxml=tr.xml test_everflow_testbed.py
test_everflow_testbed.py::TestEverflowIngressAclIngressMirror::test_everflow_case1[tor] PASSED                                                                                                              [ 16%]
test_everflow_testbed.py::TestEverflowIngressAclIngressMirror::test_everflow_case2[tor] PASSED                                                                                                              [ 33%]
test_everflow_testbed.py::TestEverflowIngressAclIngressMirror::test_everflow_case3[tor] PASSED                                                                                                              [ 50%]
test_everflow_testbed.py::TestEverflowIngressAclIngressMirror::test_everflow_case4[tor] PASSED                                                                                                              [ 66%]
test_everflow_testbed.py::TestEverflowIngressAclIngressMirror::test_everflow_case5[tor] PASSED                                                                                                              [ 83%]
test_everflow_testbed.py::TestEverflowIngressAclIngressMirror::test_everflow_case6[tor] PASSED                                                                                                              [100%]

Reply:
@daall Thanks for input. Updated accordingly.